docs/tutorials/create_your_first_library.md

```diff
@@ -0,0 +1,284 @@
+                metadata = self._read_metadata(split / "metadata.jsonl")
+
+                for view_file in split.glob("*"):
+                    # only consider {split}/{item}.{ext} files
+                    if view_file.is_file() and view_file.suffix in self.EXTENSIONS:
+                        # retrieve item metadata in metadata file
+                        item_metadata = {}
+                        for m in metadata:
+                            if m[self.view_name] == view_file.name:
+                                item_metadata = m
+                                break
+                        if not item_metadata:
+                            raise ValueError(f"Metadata not found for {view_file}")
+
+                        # extract entity metadata from item metadata
+                        entities_data = item_metadata.pop(self.entity_name, None)
+
+                        # create item
+                        item = self._create_item(split.name, **item_metadata)
+
+                        # create view
+                        view = self._create_view(item, view_file, self.view_schema)
+
+                        if entities_data is None:
+                            yield {
+                                self.item_schema_name: item,
+                                self.view_name: view,
+                            }
+                            continue
+                        elif source_id is None:
+                            source_id = self.add_source("Builder", SourceKind.OTHER)
+
+                        # create entities and their annotations
+                        entities, annotations = self._create_entities(item, view, entities_data, source_id)
+
+                        yield {
+                            self.item_schema_name: item,
+                            self.view_name: view,
+                            self.entity_name: entities,
+                            **annotations,
+                        }
+
+builder = ImageFolderBuilder(
+    source_dir=Path("./assets/health_images"),
+    target_dir=Path("./pixano_library/health_images"),
+    dataset_item=HealthDatasetItem,
+    info=DatasetInfo(
+        id="health_images",
+        name="Health Images",
+        description="A dataset of health images",
+    ),
+    url_prefix="/health_images"
+    # By default, the images' URLs are relative to the health_images folder.
+    # We assume Pixano will be launched with the "assets/" directory as media_dir.
+)
+
+dataset = builder.build(mode="create")
+```
+
+We recommend that you take a look at the implementation of the class `pixano.datasets.builders.FolderBaseBuilder` for the complete code to understand how to construct your own builder.
+
+Notice that the output of the `generate_data` consists in a dictionary whose keys are the names of the tables to fill and the values one example or a list of `pixano.features.BaseSchema` to fill these tables.
+
```

Comment:
Maybe it could be usefull to specify that we can yield data by chunks that will be appended parts by parts ? 
Or maybe not (because of consistency checks... done at the end or each yield ?)

Reply:
Consistency check is only done at the end because we don't know in what order data is put so it wouldn't make  sense to check when generating data.

I will change the sentence to say that generate_data returns an iterator because a) it is True b) it is clearer that data can be yield and not flushed completely.